### ISSUE TITLE
Examples don't work in VoiceOver

### ISSUE THREAD

**Issue by pwolfert** [downshift-js/downshift#1539]
- `downshift` version: 8.0.0, 7.x, likely others

**Relevant code or config**

[useSelect examples](https://codesandbox.io/s/github/kentcdodds/downshift-examples?file=/src/hooks/useSelect/basic-usage.js)

**What you did**:

1. Enable VoiceOver on macOS
2. View the [useSelect examples](https://codesandbox.io/s/github/kentcdodds/downshift-examples?file=/src/hooks/useSelect/basic-usage.js) in Safari
3. Open the example dropdown and navigate between items via keyboard. Notice that VoiceOver fails to read out the highlighted options

**What happened**:

VoiceOver does not seem to either recognize or correctly interpret the `aria-activedescendant` attribute, resulting in no readout when cylcing through the different options.

**Problem description**:

**Suggested solution**:

I'm afraid I don't have a simple solution, but I wanted you all to be aware and find out if you have any ideas. We've been using Downshift in a combobox pattern (with a text field) using v3.x for several years, and VoiceOver read out the options. Nowadays we have to use the a11y-message feature to fill in the missing VO messaging, but it's proven very unreliable.

A couple months ago, I read through [this article](https://zellwk.com/blog/element-focus-vs-aria-activedescendant/) about `active-descendant` and focus management, and it seemed like the more reliable strategy would be to actively move focus between the options in the listbox. One library that seems to have done that effectively is Adobe Spectrum.

That's all. I think something fundamental would have to change in how Downshift works to support that focus paradigm, but maybe ya'll have already thought about that for v9. I heard that the a11y-message system is going away.

**Comment by silviuaavram** [comment]
Hi! Can you provide Repro Steps to your issue? For me the examples work fine. I tried in Chrome.

<img width="1315" alt="Screenshot 2023-09-07 at 17 24 11" src="https://github.com/downshift-js/downshift/assets/11275392/8011e863-b893-495c-bfe9-4385420beb74">
<img width="1331" alt="Screenshot 2023-09-07 at 17 24 07" src="https://github.com/downshift-js/downshift/assets/11275392/cda12ee4-3157-4d27-8b34-368a21a69c10">
<img width="1320" alt="Screenshot 2023-09-07 at 17 24 26" src="https://github.com/downshift-js/downshift/assets/11275392/3c450eb7-dd49-4879-b8e4-9d45ad7b31b8">

**Comment by silviuaavram** [comment]
Indeed, there seems to be an issue with VO and Safari, for useCombobox. Unfortunately, the same issue occurs on the ARIA example as well. Maybe we should check the Apple VO issues.

Yes, Spectrum uses a different pattern, but the activedescendant should work as well, according to ARIA.

**Comment by silviuaavram** [comment]
Closing this due to lack of action items.
